R/bundle.R

```diff
@@ -222,15 +222,28 @@ enforceBundleLimits <- function(appDir, totalSize, totalFiles) {
 bundleApp <- function(appName, appDir, appFiles, appPrimaryDoc, assetTypeName,
                       contentCategory, verbose = FALSE, python = NULL,
                       condaMode = FALSE, forceGenerate = FALSE, quarto = NULL,
-                      isShinyApps = FALSE) {
+                      isShinyApps = FALSE, metadata = list()) {
   logger <- verboseLogger(verbose)
 
+  quartoInfo <- inferQuartoInfo(
+    appDir = appDir,
+    appPrimaryDoc = appPrimaryDoc,
+    quarto = quarto,
+    metadata = NULL
+  )
+  if (is.null(contentCategory) && !is.null(quartoInfo)) {
+    # Connect doesn't distinguish between Quarto projects and single
+    # documents, so neither will we. Quarto content are always deployed in
+    # "site" mode when using quarto::quarto_publish_app.
+    contentCategory <- "site"
+  }
+
   logger("Inferring App mode and parameters")
   appMode <- inferAppMode(
       appDir = appDir,
       appPrimaryDoc = appPrimaryDoc,
       files = appFiles,
-      quarto = quarto)
+      quarto = quartoInfo)
```

Comment:
It's unfortunate that the meaning of the `quarto` variable is not consistent; sometimes, it is a path to a binary, sometimes it is a structure. That's going to cause confusion.

Reply:
`inferAppMode` doesn't care what it receives, it just cares that it isn't `NULL`. Other than this, the changes in this PR have basically separated those two names in their usage.

Reply:
What I was trying to say: The `quarto` name sometimes represents a file path and sometimes is a `list(...)`. That means that a reader needs to remember that the type of data held by `quarto` changes from one function to another. If possible, I'd like to use a consistent name for each interpretation, especially when that data is passed in as argument.

Reply:
The interpretation is almost consistent with the changes already here. I'll change this function so that it is 100% consistent.